### PR TITLE
Catalog plugin target catalog

### PR DESCRIPTION
**Pull request by djkiourtsis**
This updates the template for displaying catalogs, and introduces the root_catalog field to the catalog plugin model.  Setting this field to a catalog will have the plugin display the contents of said catalog.

### CODE REVIEW

**Comment by djkiourtsis** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;fYwZ3q7W&#x2F;624-make-a-richcatalog-plugin-for-django-cms"> Make a richcatalog plugin for django-cms.</a>

File: shop_richcatalog/cms_plugins.py [hunk 1]
```diff
@@ -11,7 +12,16 @@ class CMSCatalogPlugin(CMSPluginBase):
     render_template = "catalog/catalog_grid_plugin.html"
 
     def render(self, context, instance, placeholder):
-        context['catalog_list'] = Catalog.objects.all()
+        context = super(CMSCatalogPlugin, self).render(
+            context, instance, placeholder)
```
Comment (mliudev) [inline shop_richcatalog/cms_plugins.py]: It doesn't look like this call to super was required before. I'm interested to know why we need it now.

**Comment by mliudev** [comment]
Code looks good to me.

File: shop_richcatalog/templates/catalog/catalog_grid_plugin.html [hunk 2]
```diff
@@ -1,21 +1,8 @@
...
-            </div>
-        </div>
-    </div>
-    {% if forloop.counter|divisibleby:4 %}</div><div class="row">{% endif %}
-    {% if forloop.last %}</div>{% endif %}
-{% endfor %}
+{% include "catalog/catalog_grid.html" with catalog_list=plugin_catalog_list %}
+{% include "product/product_grid.html" with object_list=plugin_product_list %}
+{% addtoblock 'js' %}
+  <script type="text/javascript" src="{% static 'js/equalsizer.js' %}"></script>
```
Comment (amadornimbis) [inline shop_richcatalog/templates/catalog/catalog_grid_plugin.html]: This app doesn't come bundled with equalsizer.js in its static content, and I don't think it should either. The templates for reusable Django apps shouldn't make use of any particular styling like Bootstrap, and should only include static content which it needs to function properly; these templates should be very minimal. If we want to use equalsizer.js here, then this template should be overridden in our projects.

Comment (mliudev) [inline shop_richcatalog/templates/catalog/catalog_grid_plugin.html]: @amadornimbis Nice, I hadn't thought about that. I think this is a good design pattern to follow and I second it.

File: shop_richcatalog/cms_plugins.py (hunk 1 above)
Comment (djkiourtsis) [inline shop_richcatalog/cms_plugins.py]: We don't technically need the super call, but the django cms plugin documentation suggests to populate the context this way if you are overwriting the render method.

**Pull request merged**